core/src/main/scala/org/apache/spark/scheduler/TaskLocation.scala

```diff
@@ -64,18 +64,21 @@ private[spark] object TaskLocation {
 
   /**
    * Create a TaskLocation from a string returned by getPreferredLocations.
-   * These strings have the form [hostname] or hdfs_cache_[hostname], depending on whether the
-   * location is cached.
+   * These strings have the form executor_[hostname]_[executorid], [hostname], or
+   * hdfs_cache_[hostname], depending on whether the location is cached.
    */
   def apply(str: String): TaskLocation = {
     val hstr = str.stripPrefix(inMemoryLocationTag)
     if (hstr.equals(str)) {
```

Comment:
Not related to your change, but this method could be further clarified, like by just checking upfront whether the string starts with inMemoryLocationTag and returning that case straight away. Not required here. The logic looks sound now that the comments are updated.